src/cmd/linuxkit/util.go

```diff
@@ -256,22 +256,28 @@ func NewPublishedPort(publish string) (PublishedPort, error) {
 }
 
 // CreateMetadataISO writes the provided meta data to an iso file in the given state directory
-func CreateMetadataISO(state, data string) ([]string, error) {
-	if data == "" {
+func CreateMetadataISO(state, data string, dataPath string) ([]string, error) {
+	var (
+		d   []byte
+		err error
+	)
+
+	// if we have neither data nor dataPath, nothing to return
+	switch {
+	case data == "" && dataPath == "":
 		return []string{}, nil
-	}
-
-	var d []byte
-	if st, err := os.Stat(data); os.IsNotExist(err) {
+	case data != "":
 		d = []byte(data)
-	} else if st.Size() == 0 {
-		return []string{}, nil
-	} else {
+	case dataPath != "":
 		d, err = ioutil.ReadFile(data)
 		if err != nil {
-			return nil, fmt.Errorf("Cannot read user data: %v", err)
+			return nil, fmt.Errorf("Cannot read user data from path %s: %v", dataPath, err)
+		}
+		if len(d) == 0 {
+			return []string{}, nil
```

Comment:
I wonder if in this case we ought to still create an iso with an empty file, after all that is what the user asked for.

Reply:
I was wondering about that as well; I just followed the old logic in this case, but I wonder which makes more sense. If we do, though, then shouldn't `-data ""` also get an empty file?

Reply:
> shouldn't -data "" also get an empty file?

That probably does make sense, it would mean adjusting things a bit to pass in the original flag pointer so you could distinguish nil from `""`, but that probably isn't too ugly.

Reply:
> that probably isn't too ugly

I had to open my mouth, didn't I...

Reply:
Actually, I don't think the Go flags stuff lets you distinguish `""` from "not passed at all", the pointer is not there for that purpose.

So I think we should not try and apply this to `-data ""`, it's too tricky and not important enough, people can use `-datafile /dev/null` or something if they care. See `src/cmd/linuxkit/pkglib/pkglib.go:NewFromCLI`'s use of `fs.Visit(...)` for how one can make it work (and having looked at it run away rather than implementing here ;-)).

Reply:
So then we are leaving `-data ""` as is, but changing `-data-file /some/empty-file` to have an empty metadata? Fine.